### PR TITLE
add dr auto sync

### PR DESCRIPTION
**Pull request by Liuxiaozhen12**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/6843 https://github.com/pingcap/docs-cn/pull/6886
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: two-data-centers-in-one-city-deployment.md [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+
+This section takes the example of a city where two data centers IDC1 and IDC2 are located separately in the east and west.
+
+The architecture of the cluster deployment is as follows:
+
+- The TiDB cluster is deployed to two DCs in one city: the primary IDC1 in the east, and the DR IDC2 in the west.
+- The cluster has 4 replicas, 2 Voter replicas in IDC1, 1 Voter replicas and 1 Learner replicas in IDC2. For the TiKV component, each rack has a proper label.
+- The Raft protocol is adopted to ensure consistency and high availability of data, which is transparent to users.
+
+![2-DC-in-1-city architecture](/media/two-dc-replication-1.png)
```
Comment (nolouch) [inline two-data-centers-in-one-city-deployment.md:24]: the picture is missing.

File: two-data-centers-in-one-city-deployment.md [hunk 2]
```diff
@@ -0,0 +1,242 @@
+---
+title: Two Data Centers in One City Deployment
```
Comment (nolouch) [inline two-data-centers-in-one-city-deployment.md:2]: I think it replaces the doc https://github.com/pingcap/docs/blob/master/synchronous-replication.md. maybe remove it.

File: two-data-centers-in-one-city-deployment.md (hunk 1 above)
Comment (Liuxiaozhen12) [reply two-data-centers-in-one-city-deployment.md:24]: fixed in https://github.com/pingcap/docs/pull/6182/commits/2552ec204a9da9faf59caa8ce834e9210a7fae54.

File: two-data-centers-in-one-city-deployment.md (hunk 2 above)
Comment (TomShawn) [reply two-data-centers-in-one-city-deployment.md:2]: Yes. Please help delete this file and add alias in two-data-centers-in-one-city-deployment.md. Thanks!

Comment (Liuxiaozhen12) [reply two-data-centers-in-one-city-deployment.md:2]: fixed in https://github.com/pingcap/docs/pull/6182/commits/cefc861a04d974194ae016b9bf0f124165217697.

File: two-data-centers-in-one-city-deployment.md [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+summary: Learn the deployment solution to two data centers in one city.
+aliases: ['/docs/dev/synchronous-replication-for-dual-data-centers/']
```
Comment (Liuxiaozhen12) [inline two-data-centers-in-one-city-deployment.md:4]:
```suggestion
aliases: ['/tidb/dev/synchronous-replication']
```

File: two-data-centers-in-one-city-deployment.md [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+
+### Disaster recovery
+
+This section introduces the disaster recovery solution of the two data centers in one city deployment.
+
+When a disaster occurs to a cluster in the synchronous replication mode, you can perform data recovery with `RPO = 0`:
+
+- If the primary data center fails and most of the Voter replicas are lost, but complete data exists in the DR data center, the lost data can be recovered from the DR data center. At this time, manual intervention is required with professional tools. You can contact the TiDB team for recovery.
```
Comment (nolouch) [inline two-data-centers-in-one-city-deployment.md:237]:
```suggestion
- If the primary data center fails and most of the Voter replicas are lost, but complete data exists in the DR data center, the lost data can be recovered from the DR data center. At this time, manual intervention is required with professional tools. You can contact the TiDB team for a recovery solution.
```

File: two-data-centers-in-one-city-deployment.md [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+
+- If the DR center fails and a few Voter replicas are lost, the cluster automatically switches to the asynchronous replication mode.
+
+When a disaster occurs to a cluster that is not in the synchronous replication mode and you cannot perform data recovery with `RPO = 0`:
+
+- If most of the Voter replicas are lost, manual intervention is required with professional tools. You can contact the TiDB team for recovery.
```
Comment (nolouch) [inline two-data-centers-in-one-city-deployment.md:243]:
```suggestion
- If most of the Voter replicas are lost, manual intervention is required with professional tools. You can contact the TiDB team for a recovery solution.
```

**Review by nolouch — COMMENTED** [review]
rest LGTM

File: two-data-centers-in-one-city-deployment.md [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+grafana_servers:
+  - host: 10.63.10.60
+
+alertmanager_servers:
+  - host: 10.63.10.60
+```
+
+### Placement Rules
+
+To deploy a cluster based on the planned topology, you need to use [placement rules](/configure-placement-rules.md) to determine the locations of the cluster replicas. If 4 replica and 2 Voter replica is at the primary center and 1 Voter replica and 1 Learner replica is at the DR center, you can use the placement rules to configure the replicas as follows:
```
Comment (TomShawn) [inline two-data-centers-in-one-city-deployment.md:94]:
```suggestion
To deploy a cluster based on the planned topology, you need to use [placement rules](/configure-placement-rules.md) to determine the locations of the cluster replicas. If 4 replicas and 2 Voter replicas are at the primary center and 1 Voter replica and 1 Learner replica are at the DR center, you can use the placement rules to configure the replicas as follows:
```

File: two-data-centers-in-one-city-deployment.md [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+dr-replicas = 1
+wait-store-timeout = "1m"
+wait-sync-timeout = "1m"
+```
+
+In the configuration above:
+
++ `replication-mode` is the replication mode to be enabled. In the above example, it is set to `dr-auto-sync`. By default, the majority protocol is used.
++ `label-key` is used to distinguish different data centers and needs to match placement rules. In this example, the primary data center is "east" and the DR data center is "west".
++ `primary-replicas` ia the number of Voter replicas in the primary data center.
```
Comment (TomShawn) [inline two-data-centers-in-one-city-deployment.md:191]:
```suggestion
+ `primary-replicas` is the number of Voter replicas in the primary data center.
```

File: two-data-centers-in-one-city-deployment.md [hunk 8]
```diff
@@ -0,0 +1,243 @@
...
+
+The replication mode of a cluster can automatically and adaptively switch between three statuses:
+
+- When the cluster is normal, the synchronous replication mode is enabled to maximize the data integrity of the disaster recovery data center.
+- When the network connection between the two data centers fails or the DR data center breaks down, after a pre-set protective interval, the cluster enables the asynchronous replication mode to ensure the availability of the application.
+- When the network reconnects or the DR data center recovers, the TiKV node joins the cluster again and gradually replicates the data. Finally, the cluster switches to the synchronous replication mode.
+
+The details for the status switch are as follows:
+
+1. **Initialization**: At the initialization stage, the cluster is in the synchronous replication mode. PD sends information to TiKV, and all TiKV nodes strictly follow the synchronous replication mode to work.
```
Comment (TomShawn) [inline two-data-centers-in-one-city-deployment.md:225]:
```suggestion
1. **Initialization**: At the initialization stage, the cluster is in the synchronous replication mode. PD sends the status information to TiKV, and all TiKV nodes strictly follow the synchronous replication mode to work.
```

File: two-data-centers-in-one-city-deployment.md [hunk 9]
```diff
@@ -0,0 +1,243 @@
...
+
+3. **Switch from async to sync**: PD regularly checks the heartbeat information of TiKV to judge whether TiKV nodes are reconnected. If the number of failed peers is less than the number of replicas of the primary data center (`primary-replicas`) and the DR data center (`dr-replicas`), the synchronous replication mode can be enabled again. PD first switches the status of the cluster to sync-recover and sends the status information to all TiKV nodes. All Regions of TiKV gradually switch to the two-data-center synchronous replication mode and then report the heartbeat information to PD. PD records the status of TiKV Regions and calculates the recovery progress. When all TiKV Regions finish the switching, PD switches the replication mode to sync.
```
Comment (TomShawn) [inline two-data-centers-in-one-city-deployment.md:229]:
```suggestion
3. **Switch from async to sync**: PD regularly checks the heartbeat information of TiKV to judge whether TiKV nodes are reconnected. If the number of failed nodes is less than the number of replicas of the primary data center (`primary-replicas`) and the DR data center (`dr-replicas`), the synchronous replication mode can be enabled again. PD first switches the status of the cluster to sync-recover and sends the status information to all TiKV nodes. All Regions of TiKV gradually switch to the two-data-center synchronous replication mode and then report the heartbeat information to PD. PD records the status of TiKV Regions and calculates the recovery progress. When all TiKV Regions finish the switching, PD switches the replication mode to sync.
```

File: two-data-centers-in-one-city-deployment.md [hunk 10]
```diff
@@ -0,0 +1,243 @@
...
+
+Under the mode of two data centers in one city, the two data centers are less than 50 kilometers apart. They are usually located in the same city or in two adjacent cities. The network latency between the two data centers is lower than 1.5 milliseconds and the bandwidth is higher than 10 Gbps.
+
+## Deployment architecture
+
+This section takes the example of a city where two data centers IDC1 and IDC2 are located respectively in the east and west.
+
+The architecture of the cluster deployment is as follows:
+
+- The TiDB cluster is deployed to two DCs in one city: the primary IDC1 in the east, and the DR IDC2 in the west.
```
Comment (TomShawn) [inline two-data-centers-in-one-city-deployment.md:21]:
```suggestion
- The TiDB cluster is deployed to two DCs in one city: the primary IDC1 in the east, and the disaster recovery (DR) IDC2 in the west.
```

File: two-data-centers-in-one-city-deployment.md [hunk 11]
```diff
@@ -0,0 +1,243 @@
...
+- The cluster has 4 replicas: 2 Voter replicas in IDC1, 1 Voter replicas and 1 Learner replicas in IDC2. For the TiKV component, each rack has a proper label.
```
Comment (TomShawn) [inline two-data-centers-in-one-city-deployment.md:22]:
```suggestion
- The cluster has 4 replicas: 2 Voter replicas in IDC1, 1 Voter replica and 1 Learner replica in IDC2. For the TiKV component, each rack has a proper label.
```

**Review by TomShawn — APPROVED** [review]
LGTM

File: two-data-centers-in-one-city-deployment.md [hunk 12]
```diff
@@ -0,0 +1,243 @@
...
+
+3. **Switch from async to sync**: PD regularly checks the heartbeat information of TiKV to judge whether TiKV nodes are reconnected. If the number of failed nodes is less than the number of replicas of the primary data center (`primary-replicas`) and the DR data center (`dr-replicas`), the synchronous replication mode can be enabled again. PD first switches the status of the cluster to sync-recover and sends the status information to all TiKV nodes. All Regions of TiKV gradually switch to the two-data-center synchronous replication mode and then report the heartbeat information to PD. PD records the status of TiKV Regions and calculates the recovery progress. When all TiKV Regions finish the switching, PD switches the replication mode to sync.
```
Comment (Liuxiaozhen12) [inline two-data-centers-in-one-city-deployment.md:229]:
```suggestion
3. **Switch from async to sync**: PD regularly checks the heartbeat information of TiKV to judge whether the TiKV node is reconnected. If the number of failed nodes is less than the number of replicas of the primary data center (`primary-replicas`) and the DR data center (`dr-replicas`), the synchronous replication mode can be enabled again. PD first switches the status of the cluster to sync-recover and sends the status information to all TiKV nodes. All Regions of TiKV gradually switch to the two-data-center synchronous replication mode and then report the heartbeat information to PD. PD records the status of TiKV Regions and calculates the recovery progress. When all TiKV Regions finish the switching, PD switches the replication mode to sync.
```

**Pull request merged**